packages/client-utils/src/types.d.ts

```diff
@@ -0,0 +1,33 @@
+// @file types for the client-utils package
+// NB: this doesn't follow best practices for TS in JS because this package will likely soon be written in TS
+
+import type { QuestionDetails } from '@agoric/governance/src/types.js';
+import type { MetricsNotification as VaultManagerMetrics } from '@agoric/inter-protocol/src/vaultFactory/vaultManager.js';
+import type {
+  CurrentWalletRecord,
+  UpdateRecord,
+} from '@agoric/smart-wallet/src/smartWallet.js';
+
+// For static string key types. String template matching has to be in the ternary below.
+type PublishedTypeMap = {
+  'committees.Economic_Committee.latestQuestion': QuestionDetails;
+  'auction.governance': { current: AuctionParamRecord };
+  'auction.schedule': ScheduleNotification;
+  'vaultFactory.metrics': { rewardPoolAllocation: RewardPoolAllocationRecord };
+};
+
+/**
+ * Utility type to the type that would result from unmarshalling the latest
+ * value at a vstorage `published` path.
+ */
+export type TypedPublished<T extends string> = T extends keyof PublishedTypeMap
```

Comment:
Just so I understand:  This doesn't look like a complete list of types. Is the theory that we'll gradually add types to both the list above and the nested ternary below as people come across new types they care about? Or have I missed some general case that's handling many types automatically?

Reply:
Yep, to evolve over time. Or if I have time I'll fill it out in this PR.

The ternary is only necessary for string templates. For static strings it can go in the object map type.